### ISSUE TITLE
[Question]

### ISSUE THREAD

**Issue by dutrieux** [mybatis/ehcache-cache#61]
Hello,
Is this library is supported with ehcache v3.x, don't found the information on web page : http://www.mybatis.org/ehcache-cache/

Best regards

**Comment by hazendaz** [comment]
@dutrieux This code is still based on version 2.  Nothing has been looked at as far as I'm aware for any compatibility to version 3.

**Comment by dpawlak-pgs** [comment]
any news for it?

**Comment by hazendaz** [comment]
@dpawlak-pgs I have a really rough work in progress I haven't looked at in a long time.  Its not working.  I haven't personally used ehcache3 yet and not entierly sure what all needed.  I'll push that branch up as ehcache3 which has one commit on it marked as work in progress.  Please take a look at that and see if you can try to get it working.  

Also noted, we had not released this in 2 years so I just pushed a release on the 2.x line given there were pushes from ehcache in recent time for 2.x so moving to 3.x is wide open now but needs some help.

**Comment by shiva0708** [comment]
Is there any active development to support ehcache v3 ?

**Comment by hazendaz** [comment]
Hi @shiva0708 A branch was created [ehcache3](https://github.com/mybatis/ehcache-cache/tree/ehcache3) but I never got it working.  I think I last rebased it last December.  Feel free to take a spin at it.

**Comment by ben-manes** [comment]
Unfortunately switching may be problematic. Ehcache 2.x is end-of-life so there are will be no security patches. Unfortunately, 3.x has a [severe](https://github.com/jhipster/generator-jhipster/issues/28546) denial of service attack that is easy to exploit and has been present since its pre-release. The Ehcache maintainers have said that its not a problem because users are expected to use unbounded caches so that an eviction never occurs (a full cache is user error). If you do switch, please make sure to pre-mix the hashCode to avoid introducing a vulnerability.

**Comment by ben-manes** [comment]
yep, sgtm. If you require v3.11.1+ then no mitigation is needed on your end. IBM recently took ownership as part of a larger acquisition and their security team was very responsive when I informed them. They used the FIPS certification process to ensure it was resolved.
